### PR TITLE
3d modalities

### PR DESCRIPTION
**Pull request by adavydow**

### CODE REVIEW

File: face_api_dataset/dataset.py [hunk 1]
```diff
@@ -601,9 +612,88 @@ def _open_modality(
...
+        import cv2
+
+        segments_file = (
+                self._root / f"{number}.{_modality_files(Modality.SEGMENTS)[1]}"
+        )
+        img = cv2.imread(str(segments_file), cv2.IMREAD_UNCHANGED)
+        if img is None:
+            raise ValueError(f"Error reading {segments_file}")
+        if number in self._image_sizes:
+            if self._image_sizes[number] != img.shape[::-1]:
```
Comment (Rauf-Kurbanov) [inline face_api_dataset/dataset.py:665]: I think, correct type annotation for `self._image_sizes` is currently in `main` brunch: `Dict[str, Tuple[int, int]]`

**Review by Rauf-Kurbanov — APPROVED** [review]
Minor comments

File: face_api_dataset/dataset.py [hunk 2]
```diff
@@ -601,9 +612,88 @@ def _open_modality(
...
         if modality == Modality.GAZE:
-            gaze = [
-                info["facial_attributes"]["gaze"]["horizontal_angle"],
-                info["facial_attributes"]["gaze"]["vertical_angle"],
-            ]
-            return np.array(gaze, dtype=np.float64)
+            world_to_cam = np.array(info["camera"]["transform_world2cam"]["mat_4x4"])
+            gaze_left = np.array(info["gaze_values"]["eye_left"]["gaze_vector"] + [0.0])
+            gaze_right = np.array(info["gaze_values"]["eye_right"]["gaze_vector"] + [.0])
+            print(world_to_cam.shape, gaze_left.shape)
```
[the remark this answers is not in the document]
Comment (adavydow) [reply face_api_dataset/dataset.py:618]: Thanks, removed

**Comment by adavydow** [comment]
Should not be merged before we settle on info.json futrue.

**Pull request closed without merging**